src/Magento/FunctionalTestingFramework/DataTransport/Auth/Tfa/OTP.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\DataTransport\Auth\Tfa;
+
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+use OTPHP\TOTP;
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\CredentialStore;
+
+/**
+ * Class OTP
+ */
+class OTP
+{
+    const OTP_SHARED_SECRET_PATH = 'magento/tfa/OTP_SHARED_SECRET';
+
+    /**
+     * TOTP object
+     *
+     * @var TOTP
+     */
+    private static $totp = null;
```

Comment:
Just to double check, this is caching of the shared secret, *not* the generated OTP used in the test right?

Reply:
You are right.